File: launcher/src/test/java/com/taskadapter/launcher/TALauncherTest.java
package com.taskadapter.launcher;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class TALauncherTest {
    @Test
    public void portParameterFound() {
        String[] args = {"something", "--port=9090", "something else"};
        assertEquals(9090, TALauncher.findPortNumberInArgs(args));
    }

    @Test
    public void defaultPortReturnedWhenNoParameter() {
        String[] args = {"something and something else"};
        assertEquals(9944, TALauncher.findPortNumberInArgs(args));
    }

    @Test
    public void openInBrowserArgumentDetected() {
        String[] args = {TALauncher.PARAMETER_OPEN_TASK_ADAPTER_PAGE_IN_WEB_BROWSER};
        assertTrue(TALauncher.needToOpenBrowser(args));
    }
}

File: launcher/src/test/scala/com/taskadapter/launcher/TALauncherTest.scala
package com.taskadapter.launcher

import org.junit.Assert.{assertEquals, assertTrue}
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class TALauncherTest extends FunSpec with Matchers {

  it("portParameterFound") {
    val args = Array("something", "--port=9090", "something else")
    assertEquals(9090, TALauncher.findPortNumberInArgs(args))
  }

  it("defaultPortReturnedWhenNoParameter") {
    val args = Array("something and something else")
    assertEquals(10842, TALauncher.findPortNumberInArgs(args))
  }

  it("openInBrowserArgumentDetected") {
    val args = Array(TALauncher.PARAMETER_OPEN_TASK_ADAPTER_PAGE_IN_WEB_BROWSER)
    assertTrue(TALauncher.needToOpenBrowser(args))
  }
}
